src/ppl/src/IOPlacer.cpp

```diff
@@ -795,7 +795,8 @@ int IOPlacer::assignGroupToSection(const std::vector<int>& io_group,
       }
     }
     for (auto i : sortIndexes(dst)) {
-      if (sections[i].used_slots + group_size <= sections[i].num_slots) {
+      if (sections[i].used_slots + group_size
+          <= sections[i].getMaxContiguousSlots(slots_)) {
```

Comment:
Why add used_slots here?  Isn't it enough that there are enough contiguous slots for the group?